src/views/Project/Details/General/StakeholdersModal/index.js

```diff
@@ -0,0 +1,308 @@
+    organizations.map(({ id, organization, ...others }) => (
+        { ...others, id: organization, projectOrganizationId: id }
+    ))
+);
+
+const transformOrganizationToFaramValues = (organizations = []) => (
+    listToGroupList(organizations, o => o.organizationType, o => o.organization)
+);
+
+const transformFaramValuesToOrganization = (faramValues = {}, options = []) => {
+    let organizations = [];
+    Object.entries(faramValues).forEach(([key, values]) => {
+        const organizationList = values
+            .map(v => options.find(o => o.id === v))
+            .filter(isDefined)
+            .map(({
+                id,
+                projectOrganizationId,
+                organizationDetails,
+                organizationType,
+            }) => ({
+                id: organizationType === key ? projectOrganizationId : undefined,
+                organization: id,
+                organizationType: key,
+                organizationDetails,
+            }));
+
+        organizations = [...organizations, ...organizationList];
+    });
+    return organizations;
+};
+
+function uniqueAppend(list, elements, keySelector) {
+    const map = listToMap(list, keySelector);
+    elements.forEach((el) => {
+        if (!isDefined(map[keySelector(el)])) {
+            list.push(el);
+        }
+    });
+}
+
+
+@RequestCoordinator
+@RequestClient(requestOptions)
+@FaramInputElement
+export default class StakeholdersModal extends React.PureComponent {
+    static propTypes = propTypes;
+    constructor(props) {
+        super(props);
+
+        const {
+            value,
+        } = this.props;
+
+        this.schema = {
+            fields: {
+                lead_organization: [],
+                international_partner: [],
+                national_partner: [],
+                donor: [],
+                government: [],
+            },
+        };
+
+        this.state = {
+            searchOptions: [],
+            organizationOptions: transformOrganizationToOptions(value),
+            faramValues: transformOrganizationToFaramValues(value),
+            faramErrors: {},
+        };
+    }
+
+    setOrganizationList = (newOptions) => {
+        const { organizationOptions } = this.state;
+        const transformedOptions = newOptions.map(o => ({
+            id: o.id,
+            organizationDetails: {
+                id: o.id,
+                title: o.title,
+                logo: o.logoUrl,
+            },
+        }));
+
+        const searchOptions = newOptions.map(o => ({
+            ...o,
+            logo: o.logoUrl,
+        }));
+
+        const uniqueOptions = [...organizationOptions];
+        uniqueAppend(uniqueOptions, organizationOptions, o => o.id);
+        /*
+        const uniqueOptions = unique(
+            [
+                ...transformedOptions,
+                ...organizationOptions,
+
+            ].filter(isDefined),
+            option => option.id,
+        );
+        */
```

Comment:
@frozenhelium , should we not include transformedOptions in unique options?
What was the issue with commented out code?
Also, let's avoid writing functions that mutate the values. Was the mutation necessary?